.github/workflows/image_build_push.yaml

```diff
@@ -0,0 +1,87 @@
+name: Build Image and Push to Quay
+
+on:
+  workflow_call:
+    inputs:
+      DOCKERFILE_LOCATION:
+        required: false
+        type: string
+        default: "./Dockerfile"
+      AWS_REGION:
+        required: false
+        type: string
+        default: "us-east-1"
+      AWS_ECR_REGISTRY:
+        required: false
+        type: string
+        default: "707767160287.dkr.ecr.us-east-1.amazonaws.com"
+      DOCKERFILE_BUILD_CONTEXT:
+        required: false
+        type: string
+        default: "."
+      OVERRIDE_REPO_NAME:
+        required: false
+        type: string
+        default: ""
+    secrets:
+      ECR_AWS_ACCESS_KEY_ID:
+        required: true
+      ECR_AWS_SECRET_ACCESS_KEY:
+        required: true
+      QUAY_USERNAME:
+        required: true
+      QUAY_ROBOT_TOKEN: 
+        required: true
+
+jobs:
+  ci:
+    name: Build Image and Push to Quay
+    runs-on: ubuntu-latest
+    steps:
+      # https://github.com/docker/login-action#quayio
+      - name: Login to Quay.io
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_ROBOT_TOKEN }}
+
+      # https://github.com/docker/login-action#aws-public-elastic-container-registry-ecr
+      - name: Login to ECR
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ inputs.AWS_ECR_REGISTRY }}
+          username: ${{ secrets.ECR_AWS_ACCESS_KEY_ID }}
+          password: ${{ secrets.ECR_AWS_SECRET_ACCESS_KEY }}
+        env:
+          AWS_REGION: ${{ inputs.AWS_REGION }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Set Variables
+        shell: bash
+        run: |
+          echo "IMAGE_TAG=$(echo $GITHUB_REF_NAME | sed 's/\//_/' | sed -e 's/:refs//')" >> $GITHUB_ENV
```

Comment:
based on my tests for the PRC image this doesn't work for tags

`$(echo ${GITHUB_REF#refs/*/} | tr / _)"` worked for me for both branches (refs/heads/) and tags (refs/tags/)